docs: Testing docs (#9164)
adobe/react-spectrum: packages/dev/s2-docs/pages/react-aria/CheckboxGroup/testing.mdx

import {Layout} from '../../../src/Layout';
export default Layout;

import {InlineAlert, Heading, Content} from '@react-spectrum/s2';
import testUtilDocs from 'docs:@react-aria/test-utils';
import {InstallCommand} from '../../../src/InstallCommand';
import {PatternTestingFAQ} from '../../../src/PatternTestingFAQ';

export const isSubpage = true;
export const tags = ['testing', 'checkboxgroup', 'test-utils'];
export const description = 'Testing CheckboxGroup with React Aria test utils';

# Testing CheckboxGroup

## Test utils

`@react-aria/test-utils` offers common checkbox group interaction utilities which you may find helpful when writing tests. To install, simply

> add it to your dev dependencies via your preferred package manager.

Reviewer: ```suggestion
`@react-aria/test-utils` offers common checkbox group interaction testing utilities. Install it with your preferred package manager.
```

```ts
// CheckboxGroup.test.ts
import {render} from '@testing-library/react';
import {User} from '@react-aria/test-utils';

let testUtilUser = new User({interactionType: 'mouse', advanceTimer: jest.advanceTimersByTime});
// ...

it('CheckboxGroup can select multiple checkboxes', async function () {
  // Render your test component/app and initialize the checkbox group tester
  let {getByTestId} = render(
    <CheckboxGroup data-testid="test-checkboxgroup">
    ...
    </CheckboxGroup>
  );
  let checkboxGroupTester = testUtilUser.createTester('CheckboxGroup', {root: getByTestId('test-checkboxgroup')});
  expect(checkboxGroupTester.selectedCheckboxes).toHaveLength(0);

  await checkboxGroupTester.toggleCheckbox({checkbox: 0});
  expect(checkboxGroupTester.checkboxes[0]).toBeChecked();
  expect(checkboxGroupTester.selectedCheckboxes).toHaveLength(1);

  await checkboxGroupTester.toggleCheckbox({checkbox: 4});
  expect(checkboxGroupTester.checkboxes[4]).toBeChecked();
  expect(checkboxGroupTester.selectedCheckboxes).toHaveLength(2);
});
```

> See below for the full definition of the `User` and the `CheckboxGroup` tester.

Reviewer: ```suggestion
## API
```

<ClassAPI links={testUtilDocs.links} class={testUtilDocs.exports.User} />

> <ClassAPI links={testUtilDocs.links} class={testUtilDocs.exports.CheckboxGroupTester} />

Reviewer: ```suggestion
### User

<ClassAPI links={testUtilDocs.links} class={testUtilDocs.exports.User} />

### CheckboxGroupTester

<ClassAPI links={testUtilDocs.links} class={testUtilDocs.exports.CheckboxGroupTester} />
```

also do we need to repeat the `User` API on each test page? People can also refer to it in the testing guide if needed.

Author: thats true but I felt it would be easier to have the information on each page so people didn't have to switch back and forth or if they found these testing pages via the search first

Reviewer: OK. I do think we need the headings above the ClassAPIs. It's difficult to tell what the APIs are for on the rendered page, or that there's two different ones.
